Sources/VLCActionSheet/VLCActionSheet.swift

```diff
@@ -0,0 +1,287 @@
+ *
+ * Copyright © 2018 VLC authors and VideoLAN
+ * Copyright © 2018 Videolabs
+ *
+ * Authors: Soomin Lee <bubu@mikan.io>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import Foundation
+import UIKit
+
+@objc protocol VLCActionSheetDataSource: class {
+    @objc func numberOfRows() -> Int
+    @objc func actionSheet(collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell
+}
+
+@objc protocol VLCActionSheetDelegate: class {
+    @objc optional func headerViewTitle() -> String?
+    @objc func itemAtIndexPath(_ indexPath: IndexPath) -> Any?
+    @objc func actionSheet(collectionView: UICollectionView, didSelectItem item: Any, At indexPath: IndexPath)
+}
+
+// MARK: VLCActionSheet
+class VLCActionSheet: UIViewController {
+
+    private let cellHeight: CGFloat = 50
+
+    @objc weak var dataSource: VLCActionSheetDataSource?
+    @objc weak var delegate: VLCActionSheetDelegate?
+
+    var action: ((_ item: Any) -> Void)?
+
+    lazy var backgroundView: UIView = {
+        let backgroundView = UIView()
+        backgroundView.isHidden = true
+        backgroundView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        backgroundView.backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        backgroundView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.removeActionSheet)))
+        return backgroundView
+    }()
+
+    lazy var collectionViewLayout: UICollectionViewFlowLayout = {
+        let collectionViewLayout = UICollectionViewFlowLayout()
+        collectionViewLayout.minimumLineSpacing = 1
+        collectionViewLayout.minimumInteritemSpacing = 0
+        return collectionViewLayout
+    }()
+
+    lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: UIScreen.main.bounds, collectionViewLayout: collectionViewLayout)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.backgroundColor = PresentationTheme.current.colors.background
+        collectionView.alwaysBounceVertical = true
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.register(VLCActionSheetCell.self, forCellWithReuseIdentifier: VLCActionSheetCell.identifier)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        return collectionView
+    }()
+
+    lazy var cancelButton: UIButton = {
+        let cancelButton = UIButton()
+        cancelButton.setTitle(NSLocalizedString("CANCEL", comment: ""), for: .normal)
+        cancelButton.setTitleColor(.white, for: .normal)
+        cancelButton.titleLabel?.font = UIFont.systemFont(ofSize: 16)
+        cancelButton.addTarget(self, action: #selector(self.removeActionSheet), for: .touchDown)
+        cancelButton.backgroundColor = UIColor(red:1.00, green:0.59, blue:0.13, alpha:1.0)
```

Comment:
this shouldn't be defined here where does this color come from ?

Reply:
Cancel button is not currently used, as you can see in `viewDidLoad` it's not initialized and a WIP.